packages/subscriber/src/components/add-form/categories-section.tsx

```diff
@@ -0,0 +1,133 @@
+		// Filter out any invalid category names.
+		const validTokens = tokens
+			.map( ( token ) => {
+				const tokenName = typeof token === 'string' ? token : token.value;
+				const category = newsletterCategories.find( ( cat ) => cat.name === tokenName );
+				return category?.id;
+			} )
+			.filter( ( id ): id is number => id !== undefined );
+
+		recordTracksEvent( 'calypso_subscriber_add_form_categories_change', {
+			site_id: siteId,
+			categories_count: validTokens.length,
+			action: 'added',
+		} );
+
+		setSelectedCategories( validTokens );
+	};
+
+	const handleToggle = ( value: boolean ) => {
+		setShowCategories( value );
+		if ( ! value ) {
+			setSelectedCategories( [] );
+		}
+		recordTracksEvent( 'calypso_subscriber_add_form_categories_toggle', {
+			site_id: siteId,
+			enabled: value,
+		} );
+	};
+
+	return (
+		<div className="add-subscriber__categories-container">
+			<h3>
+				{ __( 'Categories' ) } <span>({ __( 'optional' ) })</span>
+			</h3>
+			<ToggleControl
+				__nextHasNoMarginBottom
+				label={
+					<div className="categories-toggle-container">
+						<p>
+							{ createInterpolateElement(
+								__( 'Add these subscribers to specific <link>categories</link>.' ),
+								{
+									link: (
+										<a
+											href={ `/settings/newsletter/${ siteId }` }
+											target="_blank"
+											rel="noopener noreferrer"
+										/>
+									),
+								}
+							) }
+						</p>
+						<Button
+							icon={ info }
+							onClick={ () => setShowInfoPopover( ! showInfoPopover ) }
+							ref={ infoButtonRef }
+						/>
+						{ showInfoPopover && infoButtonRef.current && (
+							<Popover
+								anchor={ infoButtonRef.current }
+								onClose={ () => setShowInfoPopover( false ) }
+								position="middle left"
+								noArrow={ false }
+								ignoreViewportSize
+							>
+								<div className="categories-info-popover">
+									<p>
+										{ __(
+											'Adding newsletter categories helps you segment your subscribers more effectively.'
+										) }{ ' ' }
+										<a
+											href={ localizeUrl(
+												'https://wordpress.com/support/newsletter-settings/enable-newsletter-categories/'
+											) }
+											target="_blank"
+											rel="noopener noreferrer"
+										>
+											{ __( 'Learn more' ) }
+										</a>
+									</p>
+								</div>
+							</Popover>
+						) }
+					</div>
+				}
+				checked={ showCategories }
+				onChange={ handleToggle }
+			/>
+
+			{ showCategories && newsletterCategories && (
+				<FormTokenField
+					__next40pxDefaultSize
+					__nextHasNoMarginBottom
+					__experimentalShowHowTo={ false }
+					value={ selectedCategories
+						.map( ( id ) => newsletterCategories.find( ( cat ) => cat.id === id )?.name ?? '' )
+						.filter( ( name ) => name !== '' ) }
+					suggestions={ newsletterCategories.map( ( cat ) => cat.name ) }
+					onChange={ handleCategoryChange }
+					label=""
+					placeholder={ __( 'Search…' ) }
```

Comment:
I'm using "Search..." to match the newsletter settings, but we can iterate with a new translation to match the design.